invenio_rdm_records/services/services.py

```diff
@@ -191,3 +193,44 @@ def discard_pid(self, id_, identity, pid_type, pid_client=None):
             draft,
             links_tpl=self.links_item_tpl,
         )
+
+    def lift(self, _id, identity):
+        """Lifts embargo from the record and updates draft."""
+
+        def lift_embargo_from(rdm_record):
+            """Lifts embargo from record or draft."""
+            if not rdm_record.access.embargo.lift():
+                raise EmbargoNotLiftedError(rdm_record["id"])
+            rdm_record.access.protection.record = "public"
+            rdm_record.access.protection.files = "public"
+
+        def draft_access_field_was_modified(draft, record):
+            """Returns True if draft's access field was modified."""
+            return draft.get('access') == record.get('access')
```

Comment:
I'm not really a fan of inner functions, but this is a matter of preference 😓 what about a service level `_lift...` functions?

Reply:
Hmm, I felt like they are very specific, maybe we can move the `lift_embargo_from` to the service level since it could have some utility at some point but I would keep `draft_access_field_was_modified` as an inner function mainly because I find it very specific and its purpose is just to improve readability, if we want to get rid of this I would just remove it. Let me know if moving `lift_embargo_from` is good enough or not.